lib/swaps/SwapClientManager.ts

```diff
@@ -169,19 +169,23 @@ class SwapClientManager extends EventEmitter {
    * @returns A promise that resolves to an object containing lnd
    * clients' info, throws otherwise.
    */
-  public getLndClientsInfo = async (): Promise<LndInfos> => {
-    const lndInfos: LndInfos = {};
+  public getLndClientsInfo = async () => {
+    const lndInfos = new Map<string, LndInfo>();
     // TODO: consider maintaining this list of pubkeys
     // (similar to how we're maintaining the list of raiden currencies)
     // rather than determining it dynamically when needed. The benefits
     // would be slightly improved performance.
+    const getInfoPromises: Promise<void>[] = [];
     for (const [currency, swapClient] of this.swapClients.entries()) {
       if (isLndClient(swapClient)) {
-        lndInfos[currency] = swapClient.isDisabled()
-          ? undefined
-          : await swapClient.getLndInfo();
+        if (!swapClient.isDisabled()) {
```

Comment:
Think it's possible to use 1 conditional instead of 2 here:
`if (isLndClient(swapClient) && !swapClient.isDisabled())`

Reply:
Indeed it is, I changed it.